pygmo/_py_algorithms.py

```diff
@@ -0,0 +1,131 @@
+import random
+
+try:
+    from scipy.optimize import minimize
+
+    class scipy:
+        """
+        This class is a user defined algorithm (UDA) providing a wrapper around the function scipy.optimize.minimize.
+        The constructor accepts those arguments that are specific to the algorithm:
+        - args
+        - method
+        - tol - the tolerance
+        - callback
+        - options
+
+        Other aspects, like bounds or the existence of a gradient and hessian, are taken from the problem.
+        """
+
+        def __init__(self, args=(), method=None, tol=None, callback=None, options=None):
+            method_list = [
+                "Nelder-Mead",
+                "Powell",
+                "CG",
+                "BFGS",
+                "Newton-CG",
+                "L-BFGS-B",
+                "TNC",
+                "COBYLA",
+                "SLSQP",
+                "trust-constr",
+                "dogleg",
+                "trust-ncg",
+                "trust-exact",
+                "trust-krylov",
+            ]
+            if method in method_list + [None]:
+                self.method = method
+            else:
+                raise ValueError("Method not supported: " + method)
+
+            self.args = args
+            self.tol = tol
+            self.callback = callback
+            self.options = options
+
+        def evolve(self, population):
+            """
+            Take a random member of the population, use it as initial guess
+            for calling scipy.optimize.minimize and replace it with the final result.
+
+            Modifies the given population and returns it.
+            """
+            problem = population.problem
+
+            if problem.get_nc() > 0:
+                raise NotImplementedError(
+                    "Constraints not yet supported in SciPy wrapper, as they differ among methods."
+                )
+
+            if problem.get_nobj() > 1:
+                raise NotImplementedError("Multiple objectives not supported.")
```

Comment:
This should read something more like the usual one here: "https://github.com/esa/pagmo2/blob/master/src/algorithms/de1220.cpp#L101"

Reply:
Done